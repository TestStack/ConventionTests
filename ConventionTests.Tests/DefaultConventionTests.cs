namespace ConventionTests.Tests
{
    using System;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using AssemblyUnderTest;
    using NUnit.Framework;

    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class DefaultConventionTests
    {
        Type[] sourceTypes;

        [SetUp]
        public void Setup()
        {
            sourceTypes = typeof(SampleDomainClass).Assembly.GetTypes();
        }

        [Test]
        public void all_methods_are_virtual()
        {
            var virtualConvention = new AllMethodsAreVirtualConvention(sourceTypes);

            var exception = Assert.Throws<ConventionFailedException>(virtualConvention.AssertConvention);
            Approvals.Verify(exception.Message);
        }

        [Test]
        public void all_classes_have_default_constructor()
        {
            var constructorConvention = new ClassHasDefaultConstructorConvention(sourceTypes);

            var exception = Assert.Throws<ConventionFailedException>(constructorConvention.AssertConvention);
            Approvals.Verify(exception.Message);
        }
    }
}
