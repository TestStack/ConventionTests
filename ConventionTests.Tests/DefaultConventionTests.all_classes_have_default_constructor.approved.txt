Convention has failing items
	AssemblyUnderTest.ClassWithNoDefaultCtor does not have a default constructor
	AssemblyUnderTest.ClassWithPrivateDefaultCtor does not have a default constructor
